src/app/components/pages/home/jokes/jokes.component.ts

```diff
@@ -0,0 +1,55 @@
+import { JokeModel } from './../../../../shared/models/joke.model';
+import { JokeTypeEnum } from './../../../../shared/enums/joke-type.enum';
+import { AuthService } from './../../../../shared/services/auth.service';
+import { JokesService } from './../../../../shared/services/jokes.service';
+import { FavoriteJokeService } from './../../../../shared/services/favorite-joke.service';
+import { UserModel } from './../../../../shared/models/user.model';
+import { Component, OnInit, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-jokes',
+  templateUrl: './jokes.component.html',
+  styleUrls: ['./jokes.component.scss'],
+})
+export class JokesComponent implements OnInit {
+  @Input() showSidebar: boolean;
+  public jokes: JokeModel[];
+  public isLoading: boolean;
+  public errorMessage: string;
+  public userData: UserModel;
+
+  constructor(
+    private favoriteJokeService: FavoriteJokeService,
+    private jokesService: JokesService,
+    private authService: AuthService
+  ) {}
+
+  ngOnInit() {
+    this.jokesService.currentLoadingState.subscribe(
```

Comment:
Create some subscriber separate method for all subscribers.

Reply:
Done